### ISSUE TITLE
relationship between score decrease and inference accuracy

### ISSUE THREAD

**Issue by gieflij** [mjpost/sacrebleu#148]
Hello,
I want to know relationship between BLEU score's decrease and accuracy.
I searched about it, but I couldn't find exact answer for it.

I used WMT14 en-de checkpoints and WMT14 en-de bpe-codes.
I only changed weight data's format.

sacrebleu:
BLEU+case.lc+lang.en-de+numrefs.1+smooth.exp+test.wmt14/full+tok.13a+version.1.2.10 = **25.00** 55.8/30.7/19.2/12.4 (BP = 0.989 ratio = 0.989 hyp_len = 61999 ref_len = 62688)
BLEU+case.lc+lang.en-de+numrefs.1+smooth.exp+test.wmt14/full+tok.13a+version.1.2.10 = **27.77** 58.9/33.3/21.3/14.2 (BP = 1.000 ratio = 1.039 hyp_len = 65138 ref_len = 62688)

Here are my questions:
1. Is it right that if bleu score decreases, then inference result's accuracy is also decreased? I want to know exact relationship between BLEU score and inference accuracy.
2. Is it possible to neglect about 3 BLEU decrease in inference accuracy?
3. How many BLEU decrease is acceptable for accuracy?

**Comment by martinpopel** [comment]
What do you mean by "inference accuracy"? Exact match between the translated sentence and the reference sentence?

**Comment by gieflij** [comment]
Yes. I mean accuracy between the translated sentence after inference and the reference sentence.

**Comment by martinpopel** [comment]
It is well known that exact match between the translated sentence (after inference) and the reference translation is not a suitable metric for MT evaluation. Typically, there are many correct (and acceptable) translations, so not getting exact match of the whole sentence with a single reference does not mean the translation is wrong. This is the reason why BLEU was invented. Of course, you can compute exact-match accuracy as an additional metric - the values will be very low on the unseen test set (especially for longer sentences). It might be interesting to compute it on (a subset of) the training data to assess the amount of over-training (although even here BLEU and other MT metrics might be more suitable).

Regarding your questions:

> 1.  Is it right that if bleu score decreases, then inference result's accuracy is also decreased?

While typically, the exact-match accuracy correlates with BLEU, it does not need to be so. Imagine an almost perfect translation A, which does not have exact match with any of the sentences in the reference translation (maybe it uses synonyms and it is a perfect translation, not just almost perfect); and translation B, which is very bad except for a few short sentences where it has an exact match.

> 2. Is it possible to neglect about 3 BLEU decrease in inference accuracy?

Exact-match accuracy is not measured with BLEU.
A decrease of 3 BLEU points usually means a significant difference, but this needs to be checked with a statistical test (e.g. paired bootstrap resampling).
Even if the difference is significant, it does not need to be related to the real translation quality - a difference of 3 BLEU can be caused e.g. by just using a different style of quotation marks than the reference.

> 3. How many BLEU decrease is acceptable for accuracy?

Exact-match accuracy is not measured with BLEU.

**Comment by gieflij** [comment]
Thank you! Your answers are really helpful.
